docs/form.rst

```diff
@@ -335,17 +353,35 @@ needs a bit more info:
 
 .. php:method:: setModel($model, [$fields])
 
-Associate field with existing model object and import all editable fields
+Associate fields with existing model object and import all editable fields
 in the order in which they were defined inside model's init() method.
 
 You can specify which fields to import and their order by simply listing
 field names through second argument.
 
 Specifying "false" or empty array as a second argument will import no fields,
-so you can then use `addField` to import fields individually.
+so you can then use :php:meth:`Form::addField` to import fields individually.
+
+Note that :php:meth:`Form::setModel` also delegate adding field to the form layout
+by using `Form->layout->setModel()` internally.
 
 See also: http://agile-data.readthedocs.io/en/develop/fields.html#Field::isEditable
 
+Using setModel() on a sub layout
+--------------------------------
+
+You may add field to sub layout directly using setModel method on the sub layout itself.::
+
+    $f = $app->add('Form');
+    $f->setModel($m, false);
+
+    $sub_layout = $f->layout->addSubLayout();
+    $sub_layout->setModel($m, ['first_name', 'last_name']);
```

Comment:
can $m be optional here btw?

Reply:
also what happens if a different model is used here... ?

Reply:
Theoretically $m can be optional and then sub-layout could take model from it's parent layout ...

About different model - no idea, ave not tested that :)